src/Tools/Common/ReversedServerHelpers/ReversedServerHelpers.cs

```diff
@@ -245,6 +245,12 @@ public async Task<DiagnosticsClientHolder> Build(CancellationToken ct, int proce
                 }
                 catch (TaskCanceledException)
                 {
+                    //clean up the server
+                    if (!string.IsNullOrEmpty(fullPort) && File.Exists(fullPort))
+                    {
+                        File.Delete(fullPort);
```

Comment:
Do you need the File.Delete? I would expect that the `server.DisposeAsync()` below should clean up any resources associated with the socket. It seems strange that we would need to manually clean up any resources allocated by a pipe or socket

Reply:
To make it a little more concrete I think the change as written would introduce the following (very contrived) bug:

1. User installs dotnet-trace on their Windows machine in some folder like C:\tools\dotnet-trace
2. User creates a file in that same folder called "myfile" and they put some important information in there
3. User later runs `dotnet-trace collect --diagnostic-port myfile`
4. On Windows diagnostic-port is interpreted as a named pipe and it has nothing to do with the file in the directory that just so happens to have the exact same name. This is one of the differences between Unix Domain Sockets and Windows Named Pipes - Unix Domain Sockets create a file on the filesystem, Named Pipes do not.
5. The correct behavior is that the file would not be deleted, but with this code present it would incorrectly get deleted.

Reply:
Looking around, I think this is the case for all uses of `DiagnosticsClientHolder` that have `Dispose` called. It looks like this same snippet is there as well.

We should probably remove the `File.Delete` in both places.

Good catch.